java/clients/src/test/java/sleeper/clients/admin/IngestStatusReportScreenTest.java

```diff
@@ -0,0 +1,297 @@
+            String output = runIngestJobStatusReport()
+                    .enterPrompts(JOB_QUERY_DETAILED_OPTION, "test-job", CONFIRM_PROMPT)
+                    .exitGetOutput();
+            assertThat(output)
+                    .startsWith(CLEAR_CONSOLE + MAIN_SCREEN + CLEAR_CONSOLE)
+                    .endsWith(PROMPT_RETURN_TO_MAIN + CLEAR_CONSOLE + MAIN_SCREEN)
+                    .contains("" +
+                            "Ingest Job Status Report\n" +
+                            "------------------------\n" +
+                            "Details for job test-job");
+
+            verifyWithNumberOfInvocations(5);
+        }
+
+        @Test
+        void shouldRunIngestJobStatusReportWithQueryTypeRange() throws Exception {
+            // Given
+            createIngestJobStatusStore();
+            createSqsClient();
+            when(ingestJobStatusStore.getJobsInTimePeriod("test-table",
+                    Instant.parse("2023-03-15T14:00:00Z"), Instant.parse("2023-03-15T18:00:00Z")))
+                    .thenReturn(exampleJobStatuses());
+
+            // When/Then
+            String output = runIngestJobStatusReport()
+                    .enterPrompts(JOB_QUERY_RANGE_OPTION,
+                            "20230315140000", "20230315180000", CONFIRM_PROMPT)
+                    .exitGetOutput();
+            assertThat(output)
+                    .startsWith(CLEAR_CONSOLE + MAIN_SCREEN + CLEAR_CONSOLE)
+                    .endsWith(PROMPT_RETURN_TO_MAIN + CLEAR_CONSOLE + MAIN_SCREEN)
+                    .contains("" +
+                            "Ingest Job Status Report\n" +
+                            "------------------------\n" +
+                            "Total jobs waiting in queue (excluded from report): 10\n" +
+                            "Total jobs in defined range: 1\n");
+
+            verifyWithNumberOfInvocations(6);
+        }
+
+        private RunAdminClient runIngestJobStatusReport() {
+            return runClient().enterPrompts(INGEST_STATUS_REPORT_OPTION,
+                    INGEST_JOB_STATUS_REPORT_OPTION, "test-table");
+        }
+
+        private void createIngestJobStatusStore() {
+            InstanceProperties properties = createValidInstanceProperties();
+            setInstanceProperties(properties, createValidTableProperties(properties, "test-table"));
+            when(store.loadIngestJobStatusStore(properties.get(ID)))
+                    .thenReturn(ingestJobStatusStore);
+        }
+
+        private void createSqsClient() {
+            when(store.getSqsClient())
+                    .thenReturn(mock(AmazonSQS.class));
+            when(store.getSqsClient().getQueueAttributes(any()))
+                    .thenReturn(new GetQueueAttributesResult()
+                            .withAttributes(Map.of(
+                                    "ApproximateNumberOfMessages", "10",
+                                    "ApproximateNumberOfMessagesNotVisible", "15")));
```

Comment:
Rather than mocking the SQS client, we could mock a method to check the length of the queue. We should probably check that the name of the queue is retrieved from the instance properties.